.github/workflows/go.yaml

```diff
@@ -9,35 +9,48 @@ on:
     tags:
       - '*'
 
+env:
+  GO_VERSION: 1.22
+  GOLANGCI_LINT_VERSION: v1.59
+  STATIC_CHECK_VERSION: 2023.1.7
+  # TODO: check if it is possible to replace env.VAR on 'uses' statements on each step.
+  ACTION_IMAGE_CHECKOUT: actions/checkout@v4
+  ACTION_IMAGE_SETUP_GO: actions/setup-go@v5
+  ACTION_IMAGE_UPLOAD_ARTIFACT: actions/upload-artifact@v4
+  ACTION_IMAGE_DOWNLOAD_ARTIFACT: actions/download-artifact@v4
+  ACTION_IMAGE_SETUP_GO_FASTER: WillAbides/setup-go-faster@v1.14.0
+  ACTION_IMAGE_STATICCHECK: dominikh/staticcheck-action@v1.3.0
+  ACTION_IMAGE_GOLANGCI_LINT: golangci/golangci-lint-action@v3
```

Comment:
Those envs were used as experiment, but GHA doesn't support it explicity. I need to research how to do it. Probably it would be better to remove those envs from here in this PR.